### ISSUE TITLE
Migration fails for already setup laravel application

### ISSUE THREAD

**Issue by paschaldev** [ekmungai/eloquent-ifrs#4]
`php artisan migrate` fails 

```
Illuminate\Database\QueryException  : SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'users' already exists (SQL: create table `users` (`id` bigint unsigned not null auto_increment primary key, `entity_id` bigint unsigned null, `name` varchar(191) not null, `email` varchar(191) not null, `email_verified_at` timestamp null, `password` varchar(191) not null, `remember_token` varchar(100) null, `destroyed_at` datetime null, `deleted_at` timestamp null, `created_at` timestamp null, `updated_at` timestamp null) default character set utf8mb4 collate 'utf8mb4_unicode_ci')
```

This happens because package contains migrations that I have already setup in my application. 
`users`, `failed_job` & `password_resets` exist on an application. Can you consider removing this so an application determines their base structure.

And also consider scoping the database table table migrations. I might already have an `accounts` or `line_items` table so this would make this package difficult to use.

**Comment by ekmungai** [comment]
Thanks alot for your feedback @paschaldev. I've changed the user migration to only create the table if one doesnt exist and added scope to all other tables.

**Comment by paschaldev** [comment]
Hi @ekmungai I just saw your changes. I've been working on a fork since after the issue was created. I already made a fork and wanted to push a PR. Forgive me but I believe the fork is a little bit more dynamic.

Scenarios:
- When an app uses a users table name other than `users`: If I use `people` instead, using this library would be difficult.
- Migration class name conflicts: I already have a `transactions` migration in my app with the matching class name `CreateTransactionsTable`. This conflicted with your library too because of similar class names and db migration failed.

The fork allows the developer to pass a custom table prefix to avoid table name collisions and also pass their own custom user model. 

I removed the original `User` model in this library, stripped a few things and made it a trait instead so developers can load them themselves and implement interfaces from their own custom user model.
